mantidproject/mantid
# Adding in handling of fit numbers when fits removed

**Description of work.**

When you delete fitted data from the ADS is was causing uncaught exceptions and hard crashes as the deletion was not handled correctly. This updates the number of expected fits as data is removed to prevent this issue. The plotting is also updated so that it checks if plots have been manually removed from graphs.

**To test:**
  * Open up Muon Analysis, load some data and do a couple of fits.
  * Delete the fitted data from the ADS and check that the deletion is handled gracefully. Check that the undo fit button is also handled gracefully.
  * Plot some data and fits, try manually removing some lines and then doing some more fits, check that this is handled gracefully.



*There is no associated issue.*

*This does not require release notes* because this introduces an issue introduced in this release.


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
